BridgeAppSDK/SBABridgeManager.m

```diff
@@ -246,8 +246,25 @@ + (void)fetchChangesToScheduledActivities:(NSArray <SBBScheduledActivity *> *)sc
     }];
 }
 
+static BOOL _allowScheduleRequestsBeforeStartDate = false;
+
++ (BOOL)allowScheduleRequestsBeforeStartDate {
+    return _allowScheduleRequestsBeforeStartDate;
+}
+
++ (void)setAllowScheduleRequestsBeforeStartDate:(BOOL)allow {
+    _allowScheduleRequestsBeforeStartDate = allow;
+}
+
 + (void)fetchScheduledActivitiesFrom:(NSDate *)scheduledFrom to:(NSDate *)scheduledTo
                           completion:(SBABridgeManagerCompletionBlock)completionBlock {
+    if (!self.allowScheduleRequestsBeforeStartDate) {
+        // pin the scheduledFrom date to be no earlier than when the participant joined the study
+        NSDate *participantStartDate = SBAAppDelegate.shared.currentUser.createdOn;
+        NSComparisonResult order = [scheduledFrom compare:participantStartDate];
+        scheduledFrom = (order == NSOrderedAscending) ? participantStartDate : scheduledFrom;
```

Comment:
Yes, those test were there to check that my work-around was merging the schedule objects since it was possible to get today's schedules using both calls. They can be deleted.